### PR TITLE
Change logic with test execution

### LINKED ISSUE #59: Backend: add /executionData endpoint
Agent already is configured to POST data there; should be added accordingly.
`ExecutionData` from agent is redundant now, should be converted to `TestExecutionDto`

### PR DESCRIPTION
**Pull request by kentr0w**
### What's done:
Fixed logic
closes #59

### CODE REVIEW

File: save-common/src/commonMain/kotlin/org/cqfn/save/agent/TestExecutionDto.kt [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package org.cqfn.save.agent
+
+import kotlinx.serialization.Serializable
+import org.cqfn.save.domain.TestResultStatus
+
+@Serializable
+data class TestExecutionDto(
```
Comment (petertrr) [inline save-common/src/commonMain/kotlin/org/cqfn/save/agent/TestExecutionDto.kt:7]: We should introduce converted `TestExecution.fromDto` and `toDto` in `TestExecution` class and convert dto to entity in controller, dto shouldn't be passed to the service layer.

**Review by petertrr — CHANGES_REQUESTED** [review]

File: buildSrc/src/main/kotlin/Versions.kt [hunk 2]
```diff
@@ -21,4 +21,5 @@ object Versions {
     const val kotlinJsWrappersSuffix = "-pre.148-kotlin-1.4.30"
     const val kotlinReact = "$react$kotlinJsWrappersSuffix"
     const val jgit = "5.11.0.202103091610-r"
+    const val kotlinx = "0.1.1"
```
[the remark this answers is not in the document]
Comment (kentr0w) [reply buildSrc/src/main/kotlin/Versions.kt]: Fixed

File: buildSrc/src/main/kotlin/Versions.kt [hunk 3]
```diff
@@ -21,5 +21,5 @@ object Versions {
     const val kotlinJsWrappersSuffix = "-pre.148-kotlin-1.4.30"
     const val kotlinReact = "$react$kotlinJsWrappersSuffix"
     const val jgit = "5.11.0.202103091610-r"
-    const val kotlinx = "0.1.1"
+    const val kotlinx_datetime = "0.1.1"
```
Comment (petertrr) [inline buildSrc/src/main/kotlin/Versions.kt:24]: diktat will emit warnings on snake case naming :)

File: save-agent/src/nativeMain/kotlin/org/cqfn/save/agent/SaveAgent.kt [hunk 4]
```diff
@@ -112,7 +115,10 @@ class SaveAgent(private val config: AgentConfiguration,
                     state.value = AgentState.CLI_FAILED
                     return@coroutineScope
                 }
-                sendExecutionData(ExecutionData(emptyList()))
+                // todo: read data from files here
```
Comment (petertrr) [inline save-agent/src/nativeMain/kotlin/org/cqfn/save/agent/SaveAgent.kt:118]: This comment is not needed anymore, files are now read just above. You can change it to `parse test executions from files` though

File: save-backend/src/main/kotlin/org/cqfn/save/backend/service/TestExecutionService.kt [hunk 5]
```diff
@@ -13,24 +15,24 @@ class TestExecutionService(private val testResultRepository: TestExecutionReposi
...
     /**
-     * @param testExecutions list of test executions
+     * @param testExecutionsDto
      * @return list of lost tests
      */
-    fun saveTestResult(testExecutions: List<TestExecution>): List<TestExecution> {
-        val lostTests: MutableList<TestExecution> = mutableListOf()
-        testExecutions.forEach { testExec ->
-            val foundTestExec = testResultRepository.findById(testExec.id)
+    fun saveTestResult(testExecutionsDto: List<TestExecutionDto>): List<TestExecutionDto> {
```
Comment (petertrr) [inline save-backend/src/main/kotlin/org/cqfn/save/backend/service/TestExecutionService.kt:21]:
```suggestion
    fun saveTestResult(testExecutionsDtos: List<TestExecutionDto>): List<TestExecutionDto> {
```
Naming collections with plural names makes the code much more readable, also you won't need to abbreviate identifier inside lambda (`testExecDto`)

File: save-backend/src/main/kotlin/org/cqfn/save/backend/service/TestExecutionService.kt [hunk 6]
```diff
@@ -13,24 +15,24 @@ class TestExecutionService(private val testResultRepository: TestExecutionReposi
...
+                    this.endTime = testExecDto.endTime.toLocalDateTime()
+                    this.status = testExecDto.status
                     testResultRepository.save(this)
                 }
             },
                 {
-                    lostTests.add(testExec)
-                    log.error("Test execution with ${testExec.id} id was not found")
+                    lostTests.add(testExecDto)
+                    log.error("Test execution with ${testExecDto.id} id was not found")
```
Comment (petertrr) [inline save-backend/src/main/kotlin/org/cqfn/save/backend/service/TestExecutionService.kt:35]:
```suggestion
                    log.error("Test execution with id=[${testExecDto.id}] was not found in the DB")
```

File: save-backend/src/test/kotlin/org/cqfn/save/backend/SaveResultTest.kt [hunk 7]
```diff
@@ -44,36 +39,37 @@ class SaveResultTest {
...
             .isEqualTo<Nothing>("Some ids don't exist")
         val tests = testExecutionRepository.findAll()
-        assertFalse(tests.any { it.startTime == testExecution.startTime.withNano(0) })
-        assertFalse(tests.any { it.endTime == testExecution.endTime.withNano(0) })
+        assertFalse(tests.any { it.startTime == testExecutionDto.startTime.toLocalDateTime().withNano(0) })
+        assertFalse(tests.any { it.endTime == testExecutionDto.endTime.toLocalDateTime().withNano(0) })
+    }
+
+    companion object {
+        private val date = 1L
```
Comment (petertrr) [inline save-backend/src/test/kotlin/org/cqfn/save/backend/SaveResultTest.kt:73]: Please be more specific in variable naming, there is no way to understand what this variable means without digging into code

**Review by petertrr — CHANGES_REQUESTED** [review]

File: save-common/src/commonMain/kotlin/org/cqfn/save/agent/TestExecutionDto.kt [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+ * @property id
+ * @property agentId
+ * @property status
+ * @property startTime
+ * @property endTime
+ */
+@Serializable
+data class TestExecutionDto(
+    val id: Long,
+    var agentId: Long,
```
Comment (petertrr) [inline save-common/src/commonMain/kotlin/org/cqfn/save/agent/TestExecutionDto.kt:17]: Why do you need `var`s for these fields? I believe dto could be made immutable?

Comment (kentr0w) [reply save-common/src/commonMain/kotlin/org/cqfn/save/agent/TestExecutionDto.kt]: Fixed

**Review by petertrr — APPROVED** [review]

**Pull request merged**